internal/testutil/acc/atlas.go

```diff
@@ -0,0 +1,116 @@
+package acc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// TestMainExecution must be called from TestMain in the test package if ProjectIDExecution is going to be used.
+func TestMainExecution(m *testing.M) {
```

Comment:
can't this class also be called by multiple places in parallel?

Reply:
no, Go test calls TestMain and we call this function from there, it's like the main function but when running tests

Reply:
ok noted. I am curious to know more: `TestMain` is called once per test package. So if in a subsequent PR you also add `func TestMain(m *testing.M) {...}` to another package, will this still work? 
From how I am getting this (also based on your comment `Go runs this function before any test in the package`), it looks like there can be multiple TestMain calls from all the different test packages (which I think we have more than one)

Reply:
yeah, the idea is to add TestMain to all packages.
when you run tests you choose the package, and as TestMain is a public function (uppercase) there can be only one TestMain for that package, and that's the function to be called when running the test for that package.

So it can't happen that two TestMain are run in a test execution,   the execution is only for one package, and that can have at most one TestMain